src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -155,7 +155,13 @@ define(function (require, exports, module) {
          * @param {string} newName 
          */
         _handleLayerNameChange: function (event, newName) {
-            this.getFlux().actions.layers.rename(this.props.document, this.props.layer, newName);
+            if (newName.length !== 0) {
+                if (newName !== this.props.layer.name) {
+                    this.getFlux().actions.layers.rename(this.props.document, this.props.layer, newName);
+                }
+            } else {
+                this.refs.layerName.setValue(this.props.layer.name);
```

Comment:
for the sake of conversation... I know what you mean @volfied.  I wonder about an alternate approach:  If the onChange handler (`_handleLayerNameChange` in this case) determines that the changed text value is invalid (empty, in this case), then it could just force *itself* to re-render based on on its current props/state... which would in turn cause TextInput to reset back to a steady state.